src/datepicker/datepicker.js

```diff
@@ -220,7 +220,16 @@ export default class Datepicker extends React.Component<
       }
     } else {
       const dateString = this.normalizeDashes(inputValue);
-      let date = new Date(dateString);
+      let yearString = dateString.split('/')[0].trim();
```

Comment:
How does it handle i18n cases or custom formatting?

Reply:
My premise here is that datepicker works well with format with "/" delimiter. 
This change is applied to any datestring where the length of the first subset is less than 4 
so for all the date formats with YYYY as the first format of the date, you could see the the change in the popup calendar UI. 
For the custom formats which starts with MM or DD, this change would be applied but the formatCheck would make the date as null so you wouldn't see any change there. 

If there is a cleaner way to do this, do let me know.

Reply:
For i18n, 
I noticed that localisation works with the calender input not the manual entry.Thought this as bug so didnt want to include in this PR.  The inputted date doesn't reflected in the calendar UI. 
![ezgif-6-89d40bdcc8c5](https://user-images.githubusercontent.com/9390424/79880931-3f9ec600-840e-11ea-8487-467345096d27.gif)

If i am missing any cases or context here let me know.

Reply:
@nadiia  formatString property is actually not triggering this behavior.